reporting/transformer_example/src/transformer_example.py

```diff
@@ -50,82 +39,82 @@ class ElasticsearchRecord(object):
-    s3_record = _get_record_from_s3(s3, hybrid_record.location)
-    return ElasticsearchRecord(id=hybrid_record.id, doc=s3_record)
+    Returns
+    -------
+    data_dict : dict
+        object data in malleable dict format
+    """
+    object_location = hybrid_record.location
+    s3_object = s3.get_object(Bucket=object_location.namespace, Key=object_location.key)
+    record = s3_object["Body"].read().decode("utf-8")
+    record_data = json.loads(record)["data"]
+    return record_data
 
 
 def extract_records(s3, event):
+    """
+    extracts records from VHS and prepares them for transformation
+
+    Parameters
+    ----------
+    s3 : boto3.client
+    event : dict
+
+    Returns
+    -------
+    data_dicts : dict
+        a list of records in dict form, as specified by the input event
+    """
     raw_hybrid_records = [
         json.loads(record["Sns"]["Message"]) for record in event["Records"]
     ]
-    hybrid_records = [
-        _extract_hybrid_record(raw_record) for raw_record in raw_hybrid_records
+    hybrid_records = [HybridRecord(**record) for record in raw_hybrid_records]
+    data_dicts = [
+        hybrid_record_to_data_dict(s3, hybrid_record)
+        for hybrid_record in hybrid_records
     ]
-    return [_build_es_record(s3, hybrid_record) for hybrid_record in hybrid_records]
-
+    return data_dicts, hybrid_records
 
-def transform_records(records):
-    return [_transform(record) for record in records]
-
-
-def _load_record(es, index, doc_type, es_record):
-    return es.index(index=index, doc_type=doc_type, id=es_record.id, body=es_record.doc)
-
-
-def load_records(es, index, doc_type, docs):
-    return [_load_record(es, index, doc_type, doc) for doc in docs]
-
-
-def _run(elasticsearch_client, event, index, doc_type, s3_client):
-    records = extract_records(s3_client, event)
-    transformed_records = transform_records(records)
-    return load_records(elasticsearch_client, index, doc_type, transformed_records)
-
-
-def elasticsearch_client(url, username, password):
-    return Elasticsearch(
-        url, http_auth=(username, password), use_ssl=True, ca_certs=certifi.where()
-    )
-
-
-def main(
-    event,
-    _,
-    s3_client=None,
-    elasticsearch_client=None,
-    elasticsearch_index=None,
-    elasticsearch_doctype=None,
-):
-    print(f"Event: {event}")
 
+# Move records with transforms applied -----------------------------------------
+def main(event, _, s3_client=None, es_client=None, index=None, doc_type=None):
+    """
+    get records from VHS, apply the transformation to them, and shove them into
+    an elasticsearch index
+    """
     s3_client = s3_client or boto3.client("s3")
-
-    elasticsearch_client = elasticsearch_client or elasticsearch_client(
-        url=os.environ["ES_URL"],
-        username=os.environ["ES_USER"],
-        password=os.environ["ES_PASS"],
+    index = index or os.environ["ES_INDEX"]
+    doc_type = doc_type or os.environ["ES_DOC_TYPE"]
```

Comment:
👍 for moving this stuff up, and getting environment parsing out of the way early.